### PR TITLE
[ML] Returns multiple models for GET _inference

### PR DESCRIPTION
**Pull request by davidkyle**
Changes GET inference models to return a list of models.

```
{
  "models": [
    {
      "model_id": "foo"
     ...
  ]
}
```

This is a non-issue as wildcard get is not supported yet but the change is necessary for the clients to have the correct format.

### CODE REVIEW

**Review by jonathan-buttner — APPROVED** [review]

**Comment by davidkyle** [comment]
@elasticmachine test this please

**Comment by davidkyle** [comment]
@elasticmachine update branch

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/IdleConnectionEvictor.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public synchronized void start() {
     private void startInternal() {
         logger.debug(() -> format("Idle connection evictor started with wait time: [%s] max idle: [%s]", sleepTime, maxIdleTime));
 
-        Scheduler.Cancellable task = threadPool.scheduleWithFixedDelay(() -> {
+        cancellableTask.set(threadPool.scheduleWithFixedDelay(() -> {
```
Comment (davidkyle) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/IdleConnectionEvictor.java:70]: I hit a unit test failure where `IdleConnectionEvictor::close` was called but the scheduled action before the task had been set.

**Comment by davidkyle** [comment]
@elasticmachine update branch

**Pull request merged**